### PR TITLE
Docscleanup

### PR DESCRIPTION
**Pull request by loriab**
## Description
Docs cleanup

## Todos
Notable points that this PR has either accomplished or will accomplish.
* **Developer Interest**
  - [x] allows building docs w/o psi-sphinx-theme (though you lose psivar indexing)
  - [x] makes sure all the tests (incl. psiapi/json) get copied to samples
  - [x] fixes up misc. options-referencing errors

## Checklist
- [x] built with alabaster sphinx theme

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: doc/sphinxman/document_tests.pl [hunk 1]
```diff
@@ -60,6 +60,15 @@
...
+   "v2rdm_casscf/" => "v2rdm_casscf",
+   "snsmp2/"    => "snsmp2",
+   "simint/"    => "simint",
+   "gpu_dfcc/"  => "gpu_dfcc",
+   "gcp/"       => "gcp",
+   "cookbook/"  => "cookbook",
+   "json/"      => "json",
+   "psi4numpy/" => "psi4numpy",
+   "python/"    => "python",
 );
```
Comment (andysim) [inline doc/sphinxman/document_tests.pl]: didn't realize the good ol' perl script is still handling this.  blast from the past!

File: samples/json/schema-1-energy/input.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  "return_energy": expected_return_result
+}
+
+# Switch run to exact MP2
+json_data["keywords"]["scf_type"] = "pk"
+json_data["keywords"]["mp2_type"] = "conv"
+json_ret = psi4.json_wrapper.run_json(json_data)
+
+
+for k in expected_properties.keys():
```
Comment (andysim) [inline samples/json/schema-1-energy/input.py]: Maybe the corresponding test input needs a `#TEST` label, so prevent this null loop being generated

Comment (andysim) [reply samples/json/schema-1-energy/input.py]: I'd be happy to use the magic edit tool, but that won't help here - the samples will need regeneration.

File: samples/dft-grad-lr1/test.in [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+set {
+    scf_type              df
+    basis                 cc-pvdz
+    dft_radial_points     99
+    dft_spherical_points  302
+    points 5
+}
+
+anl_grad = gradient('wB97X-D', dertype=1)
+compare_matrices(df_ref, anl_grad, 7, "DF Analytic Gradient vs Reference")    #TEST
```
[the remark this answers is not in the document]
Comment (loriab) [reply samples/dft-grad-lr1/test.in]: they're stripped out of `input.py` & `input.dat`. there's actually two copies now, where `test.dat` or `.py` retains the `compare_values()` lines, just so one has something of an installed test suite.

[the remark this answers is not in the document]
Comment (andysim) [reply samples/dft-grad-lr1/test.in]: Ah, ok.  Thanks for the clarifaction

[the remark this answers is not in the document]
Comment (andysim) [reply samples/dft-grad-lr1/test.in]: ... and the clarification

File: samples/json/schema-1-energy/input.py (hunk 2 above)
Comment (andysim) [reply samples/json/schema-1-energy/input.py]: Great, thanks.  I will flag any more missing `#TEST` that I see but LGTM
